scala2java-core/src/main/scala/io/github/effiban/scala2java/core/extensions/ExtensionRegistryBuilder.scala

```diff
@@ -3,15 +3,23 @@ package io.github.effiban.scala2java.core.extensions
 import io.github.effiban.scala2java.spi.Scala2JavaExtension
 
 import java.util.ServiceLoader
-import scala.jdk.CollectionConverters._
+import scala.jdk.StreamConverters._
+import scala.meta.Source
 
-object ExtensionRegistryBuilder {
+class ExtensionRegistryBuilder {
 
-  def build(): ExtensionRegistry = {
-    val extensions = ServiceLoader.load(classOf[Scala2JavaExtension])
-      .iterator
-      .asScala
+  def buildFor(source: Source): ExtensionRegistry = {
+    val extensions = loadExtensions()
+      .map(_.get)
+      .filter(_.shouldBeAppliedTo(source))
       .toList
     ExtensionRegistry(extensions)
   }
+
+  // Visible for testing
+  private[extensions] def loadExtensions() = ServiceLoader.load(classOf[Scala2JavaExtension])
+    .stream()
+    .toScala(LazyList)
```

Comment:
The current implementation is not really lazy, since it always `get`s on line 13 - however in order to remain flexible, I prefer to do it like this in case in th efuture we wish to add some filter which can be evalutaed without instantiating the extension (e.g. by reading annotations).